src/analysis/calculus/fderiv_measurable.lean

```diff
@@ -0,0 +1,455 @@
+    -- (in fact, we use `m = k - 1` instead of `k` because of the precise definition of `A`).
+    have J1 : ∥f (x + y) - f x - L e (n e) m ((x + y) - x)∥ ≤ (1/2) ^ e * (1/2) ^ m,
+    { apply le_of_mem_A (hn e (n e) m (le_refl _) m_ge).2.2,
+      { simp only [mem_closed_ball, dist_self],
+        exact div_nonneg (le_of_lt P) (zero_le_two) },
+      { simp [dist_eq_norm],
+        convert h'k,
+        field_simp,
+        ring_exp } },
+    have J2 : ∥f (x + y) - f x - L e (n e) m y∥ ≤ 4 * (1/2) ^ e * ∥y∥ := calc
+      ∥f (x + y) - f x - L e (n e) m y∥ ≤ (1/2) ^ e * (1/2) ^ m :
+        by simpa only [add_sub_cancel'] using J1
+      ... = 4 * (1/2) ^ e * (1/2) ^ (m + 2) : by { field_simp, ring_exp }
+      ... ≤ 4 * (1/2) ^ e * ∥y∥ :
+        mul_le_mul_of_nonneg_left (le_of_lt hk) (mul_nonneg (by norm_num) (le_of_lt P)),
+    -- use the previous estimates to see that `f (x + y) - f x - f' y` is small.
+    calc ∥f (x + y) - f x - f' y∥
+        = ∥(f (x + y) - f x - L e (n e) m y) + (L e (n e) m - f') y∥ :
+      by { congr' 1, simp, abel }
+    ... ≤ ∥f (x + y) - f x - L e (n e) m y∥ + ∥(L e (n e) m - f') y∥ :
+      norm_add_le _ _
+    ... ≤ 4 * (1/2) ^ e * ∥y∥ + 12 * ∥c∥ * (1/2) ^ e * ∥y∥ :
+      add_le_add J2
+        (le_trans (le_op_norm _ _) (mul_le_mul_of_nonneg_right (Lf' _ _ m_ge) (norm_nonneg _)))
+    ... = (4 + 12 * ∥c∥) * ∥y∥ * (1/2) ^ e : by ring
+    ... ≤ (4 + 12 * ∥c∥) * ∥y∥ * (ε / (4 + 12 * ∥c∥)) :
+      mul_le_mul_of_nonneg_left he.le
+        (mul_nonneg (add_nonneg (by norm_num) (mul_nonneg (by norm_num) (norm_nonneg _)))
+          (norm_nonneg _))
+    ... = ε * ∥y∥ : by { field_simp [ne_of_gt pos], ring } },
+  rw ← this.fderiv at f'K,
+  exact ⟨this.differentiable_at, f'K⟩
+end
+
+theorem differentiable_set_eq_D (hK : is_complete K) :
+  {x | differentiable_at 𝕜 f x ∧ fderiv 𝕜 f x ∈ K} = D f K :=
+subset.antisymm (differentiable_set_subset_D _) (D_subset_differentiable_set hK)
+
+end fderiv_measurable_aux
+
+open fderiv_measurable_aux
+
+lemma fderiv_mem_iff {f : E → F} {s : set (E →L[𝕜] F)} {x : E} :
+  fderiv 𝕜 f x ∈ s ↔ (differentiable_at 𝕜 f x ∧ fderiv 𝕜 f x ∈ s) ∨
+    (0 : E →L[𝕜] F) ∈ s ∧ ¬differentiable_at 𝕜 f x :=
+begin
+  split,
+  { intro hfx,
+    by_cases hx : differentiable_at 𝕜 f x,
+    { exact or.inl ⟨hx, hfx⟩ },
+    { rw [fderiv_zero_of_not_differentiable_at hx] at hfx,
+      exact or.inr ⟨hfx, hx⟩ } },
+  { rintro (⟨hf, hf'⟩|⟨h₀, hx⟩),
+    { exact hf' },
+    { rwa [fderiv_zero_of_not_differentiable_at hx] } }
+end
+
+variables [measurable_space E] [opens_measurable_space E]
+variables (𝕜 f)
+
+/-- The set of differentiability points of a function, with derivative in a given complete set,
+is Borel-measurable. -/
+theorem is_measurable_set_of_differentiable_at_of_is_complete
+  {K : set (E →L[𝕜] F)} (hK : is_complete K) :
+  is_measurable {x | differentiable_at 𝕜 f x ∧ fderiv 𝕜 f x ∈ K} :=
+begin
+  rw differentiable_set_eq_D K hK,
+  refine is_measurable.Inter (λ e, _),
```

Comment:
Should we turn `is_measurable.Inter` etc into `simp` lemmas so that this will become `simp [is_open_B.is_measurable]`?

Reply:
For now, I have just used them as simp lemmas in the proof, but I haven't marked them as simp lemmas globally.